walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,42 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.sql.Time;
+import java.util.*;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    SortedSet<TimeRange> timeRangesSorted = new TreeSet<TimeRange>(TimeRange.ORDER_BY_START);
+    for(Event e: events){ //some map could be used here
+      if(hasAttendee(e, request.getAttendees())){
+        timeRangesSorted.add(e.getWhen());
+      }
+    }
+
+    TimeRange guardian = TimeRange.fromStartDuration(TimeRange.END_OF_DAY+1, 0); //Shouldn't timerange set outside the minutes of the day be invalid?
+    timeRangesSorted.add(guardian); //Could be done by an additional if at the end as well
```

Comment:
Tough question. I see that the guardian here makes the main loop much simpler. But it can be considered invalid. And while technically it is not invalid - at the moment - we can imagine that in the future the constructor can throw an exception if the time is outside of the day zone.
Two options here:
1. Drop the guardian and do the if, or
2. In the spirit of this week topic, add a test (in a proper test file) that asserts that such a TimeRange can be created (and maybe also checking that duration between START_OF_DAY and such a guardian, exclusive, is equal to the whole day). This way, if someone in the future would like to add validation to the constructor, unit tests for TimeRange would fail and the test case would explain why such guardian is needed.

Reply:
I have put the test in the TimeRangeTest class. Is that alright?